### PR TITLE
dependencies added, created index.js, server.js, knexfile.js and data…

### PR DESCRIPTION
**Pull request by harjothkhara**
… folder.

### CODE REVIEW

File: data/helpers/user-model.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+async function update(item, id) {
+    const editedUser = await db("users")
+        .where({ id })
+        .update( item );
+    if (editedUser) {
+        const user = await findById(id);
+        return user;
+    }
+}
```
Comment (angel-torres) [inline data/helpers/user-model.js]: ✔️

File: data/migrations/20190610121207_users.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+            .unique();
+        table.string("password", 30).notNullable();
+        table.string("first_name", 20);
+        table.string("last_name", 20);  
+    });
+};
+
+exports.down = function(knex, Promise) {
+  return knex.schema.dropTableIfExists("users");
+};
```
Comment (angel-torres) [inline data/migrations/20190610121207_users.js]: ✔️

File: data/seeds/users.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  console.log(fakeUsers);
+
+  fs.writeFileSync("./authTestInfo.json", JSON.stringify({ users: fakeUsers }));
+
+  fakeUsers.map(user => {
+    user.password = bcrypt.hashSync(user.password, 4);
+  });
+
+  return await knex("users").insert(fakeUsers);
+};
```
Comment (angel-torres) [inline data/seeds/users.js]: That's soo cool you were able to use faker to make fake users! 🔥

File: routers/loginRouter.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+            }
+        } catch (error) {
+            res.status(500).json({ message: `Your user could not be created ${error}.` });
+        }
+    }
+
+});
+
+
+module.exports = router;
```
Comment (angel-torres) [inline routers/loginRouter.js]: ✔️

**Review by angeltorgon — COMMENTED** [review]
## Minimum Viable Product Objectives
* [x] POST endpoint that creates a user and hashes password
* [x] POST endpoint that authenticates user
* [x] GET endpoint that returns all users if user if logged in

## Stretch Objectives
* [ ] Add middleware that insures a user is logged in when accessing routes

# Project Manager Feedback
## Great
* You completed all MVP objectives! :heavy_check_mark:
Awesome work Harjoth!
## Requested Improvements
* Try out some stretch objectives!
## Questions
* How comfortable do you feel authenticating a user?
## Rating {2}
:star: :star:

**Pull request merged**